CHANGELOG.md

```diff
@@ -0,0 +1,16 @@
+# Changelog
+All notable changes to this project will be documented in this file.
```

Comment:
Just curious, why is this documentation necessary versus looking at PRs which explicitly in the title talks about the work that happened? 

Also, what would qualify as notable changes? Just looking for examples and need for this. :)

Reply:
If you haven't already done so, I recommend reading all of https://keepachangelog.com/en/1.0.0/. It talks about some of the philosophy. 

Notable changes IMO are functional differences, that fall into one of the categories of `Changed`, `Added`, `Removed`. Sometimes explicitly adding a `Breaking Changes` is nice. 

Pull requests often have more detail that is needed or desired for a potential user of some tool. They just need to know at a high level what features have been added and don't care about the technical details (but if they want them - they can go to the pull request). So the Changelog is really a form of documentation that has a different target audience from pull requests.

Reply:
Sounds fair! What's the best practice to remember to add these change logs?

Reply:
Heh, "Remember to Update them". In the past I've seen PR workflows (pipelines) added to check that the Changelog is edited as a part of each PR. But that's too heavy handed IMO. Not everything well warrant it. Like the PR Rick just merged, nothing ChangeLog worthy there, just house keeping. Also Review.

Reply:
This is a great question, @shalinikhare27. I think @kyle-rader's done a pretty good job of explaining what this is for. The main audience is our users. Rather than expecting them to sift through PRs or commits between versions we can spell out all the important changes in a single place. Then, when they upgrade from `v0.1.0` to `v0.2.0` (or whatever) they can just read `CHANGELOG.md`. Other projects often just call this file `HISTORY`, but the format @kyle-rader suggested here provides a good framework for determining what changes are impactful enough to be documented.

As for remembering to update this changelog, like @kyle-rader said, you can automate some checks. I've done that in the past before and it because a pain really quickly. Would not recommend it! 🙅‍♂️ Instead, for now, I think a best effort approach is good enough and we can re-evaluate if that becomes problematic.